File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpContainerMoveBatchData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpContainerMoveBatchData(
        String format,
        boolean dryRun,
        boolean stopOnError,
        int requestedMoves,
        int succeededMoves,
        int failedMoves,
        int totalMovedCount,
        List<Result> results
) {
    public record Result(
            int index,
            String code,
            int requestedCount,
            int movedCount,
            RMcpContainerData.Slot movedItem,
            RMcpContainerMoveData.Endpoint from,
            RMcpContainerMoveData.Endpoint to,
            RMcpContainerData fromContainer,
            RMcpContainerData toContainer
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpContainerMoveBatchData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpContainerMoveBatchData(
    val action: String,
    val failedMoves: List<FailedMove>
) {
    @JvmRecord
    data class FailedMove(
        val index: Int,
        val code: String?
    )
}
